src/js/utils/permissionUtils.js

```diff
@@ -41,4 +43,12 @@ const canEditRequest = (currentUser, request, location) => {
   return isLocationDestination && isUserRequestor;
 };
 
+// To have permissions for product source list actions
+// user has to be at least admin and has product manager permissions
+export const hasPermissionsToProductSourceActions = (user) => {
+  const hasProductManagerPermission = hasRole(user, RoleType.ROLE_PRODUCT_MANAGER);
+  const isAdmin = hasRole(user, RoleType.ROLE_ADMIN) || hasRole(user, RoleType.ROLE_SUPERUSER);
```

Comment:
you have this part available in `getAppContext` (isUserAdmin).

Reply:
To be honest, I am not a big fan of getting all of those:  isUserAdmin, isUserManager, etc. from the getAppContext, I think it should be calculated on the frontend part based on the users' roles. For me it is a little bit different when I have a function `hasPermissionsTo...` and I am passing, let's say, half of the things that should be checked within this function (In this example isAdmin should be passed from getAppContext, but I think it is a purpose of this function - to check if user has admin role and if has a product manager permission). But If the `isAdmin` is available from the `getAppContext`, I am going to change it.

Reply:
neither am I, it probably needs a deeper refactor, but just wanted to mention that we have it available.
I like your idea of adding `roles` property for the user dto and I hope this will go through.